webapp/javascript/pages/loading.ts

```diff
@@ -0,0 +1,7 @@
+export function isLoadingOrReloading(
+  states: ('pristine' | 'loading' | 'reloading' | 'loaded')[]
+) {
+  return states.every((state) => {
```

Comment:
[].every means `true` if all array items return `true`, just a question, didn't you consider returning `true` from this function using [].some? in my point of view, if component requires data from 2 sources, and one of them is loading, it means there is no enough data for rendering this component, so this component is in `loading` state

Reply:
Good point! Updated!